app/validation/validator.py

```diff
@@ -169,14 +174,23 @@ def _ensure_all_variant_ids_consistent(self, block, variants):
 
             for answer in question_variant['answers']:
                 answer_ids.add(answer['id'])
+                answer_types.add(answer['type'])
 
         if len(question_ids) != 1:
             errors.append(self._error_message(
                 'Variants contain more than one question_id for block: {}. Found ids: {}' .format(block['id'], question_ids)))
 
+        if len(question_types) != 1:
+            errors.append(self._error_message(
+                'Variants have more than one question type for block: {}. Found types: {}'.format(block['id'], question_types)))
+
+        if len(answer_types) != 1:
```

Comment:
I'm not sure this will work if a single variant has multiple answers with different types?

e.g. all variants have two answers with types number and textfield.

Reply:
Good point, certainly possible, but I don't think I have ever seen a question with 2 different answer types.